### PR TITLE
Support Veracode code scanning

### PR DESCRIPTION
**Pull request by mdesmet**
## Overview

Support Veracode code scanning

## Checklist

- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] `README.md` updated and added information about my change
- [x] `CHANGELOG.md` updated and added information about my change

### CODE REVIEW

**Comment by mdesmet** [comment]
tested using `debug: true`

```
Run veracode/veracode-sca@v1.09
Start command
curl -sSL https://download.sourceclear.com/ci.sh | sh -s -- scan . --debug  --json=scaResults.json
Finish command
Scan finished with exit code:  0
getVeracodeLabel - START
Finish command
=======================   ERROR   ===============================
RequestError [HttpError]: Not Found
    at /home/runner/work/_actions/veracode/veracode-sca/v1.09/dist/index.js:3694:21
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  status: 404,
  response: {
    url: 'https://api.github.com/repos/mdesmet/dbt-trino/labels/Veracode%20Dependency%20Scanning',
    status: 404,
    headers: {
      'access-control-allow-origin': '*',
      'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset',
      connection: 'close',
      'content-encoding': 'gzip',
      'content-security-policy': "default-src 'none'",
      'content-type': 'application/json; charset=utf-8',
      date: 'Thu, 04 Aug 2022 16:51:59 GMT',
      'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
      server: 'GitHub.com',
      'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
      'transfer-encoding': 'chunked',
      vary: 'Accept-Encoding, Accept, X-Requested-With',
      'x-content-type-options': 'nosniff',
      'x-frame-options': 'deny',
      'x-github-media-type': 'github.v3; format=json',
      'x-github-request-id': '0440:704A:6E88A0:1267725:62EBF92F',
      'x-ratelimit-limit': '1000',
      'x-ratelimit-remaining': '999',
      'x-ratelimit-reset': '16596[35](https://github.com/mdesmet/dbt-trino/runs/7677005402?check_suite_focus=true#step:3:36)519',
      'x-ratelimit-resource': 'core',
      'x-ratelimit-used': '1',
      'x-xss-protection': '0'
    },
    data: {
      message: 'Not Found',
      documentation_url: 'https://docs.github.com/rest/reference/issues#get-a-label'
    }
  },
  request: {
    method: 'GET',
    url: 'https://api.github.com/repos/mdesmet/dbt-trino/labels/Veracode%20Dependency%20Scanning',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'octokit-core.js/3.5.1 Node.js/12.22.7 (linux; x64)',
      authorization: 'token [REDACTED]'
    },
    request: { agent: [Agent], hook: [Function: bound bound register] }
  }
}
getVeracodeLabel - END
createVeracodeLabels - END
createVeracodeLabels - END
getIssues - START
getIssues - END
Checking for issue title [CVE: 0000-0000 found in Jinja2 - Version: 2.11.3 [PYTHON]]
Checking for issue title [CVE: 2020-14343 found in PyYAML - Version: 5.3.1 [PYTHON]]
Error: Found Vulnerability with CVSS equal or greater than 1
Scan finished.
Full Report Details:   https://sca.analysiscenter.veracode.com/teams/000tO0Y/scans/[39](https://github.com/mdesmet/dbt-trino/runs/7677005402?check_suite_focus=true#step:3:40)67[57](https://github.com/mdesmet/dbt-trino/runs/7677005402?check_suite_focus=true#step:3:58)[81](https://github.com/mdesmet/dbt-trino/runs/7677005402?check_suite_focus=true#step:3:82)
```

File: .github/workflows/security.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+on: 
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - "**/*.md"
+  schedule:
+    - cron: 15 14 * * 6
```
Comment (hovaesco) [inline .github/workflows/security.yml:8]: what do you say about `00 1 * * 1` schedule, we will have fresh result on mondays.

File: .github/workflows/security.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+jobs:
+  veracode-sca-task:
+    runs-on: ubuntu-latest
+    name: Scan repository for Issues
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Run Veracode SCA
```
Comment (hovaesco) [inline .github/workflows/security.yml:18]: Where are the results being stored?

File: .github/workflows/security.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        env:
+          SRCCLR_API_TOKEN: ${{ secrets.SRCCLR_API_TOKEN }}
+        uses: veracode/veracode-sca@v1.09
+
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          quick: true
+          create-issues: true 
+          fail-on-cvss: 1
```
Comment (hovaesco) [inline .github/workflows/security.yml:27]: `fail-on-cvss` is `The maximum allowed cvss in found vulnerabilities to pass the step` so should it be 0?

File: .github/workflows/security.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    runs-on: ubuntu-latest
+    name: Scan repository for Issues
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Run Veracode SCA
+        env:
+          SRCCLR_API_TOKEN: ${{ secrets.SRCCLR_API_TOKEN }}
+        uses: veracode/veracode-sca@v1.09
```
Comment (hovaesco) [inline .github/workflows/security.yml:21]: Does it scan dependencies as well?

**Review by hovaesco — COMMENTED** [review]
Test failed with:
```
RequestError [HttpError]: Not Found
    at /home/runner/work/_actions/veracode/veracode-sca/v1.09/dist/index.js:3694:21
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

File: .github/workflows/security.yml (hunk 2 above)
Comment (mdesmet) [reply .github/workflows/security.yml:18]: The results are created as Github issues.

File: .github/workflows/security.yml (hunk 3 above)
Comment (mdesmet) [reply .github/workflows/security.yml:27]: You are right

File: .github/workflows/security.yml (hunk 4 above)
Comment (mdesmet) [reply .github/workflows/security.yml:21]: It does: see https://docs.veracode.com/r/c_sc_what_is

**Review by alizastarburst — APPROVED** [review]
This looks fine to me

File: .github/workflows/security.yml (hunk 3 above)
Comment (mdesmet) [reply .github/workflows/security.yml]: IMHO we should not fail the build. We should just create issues.

As you can see, both of these issues are not actionable by us as they are dbt dependencies.

Checking for issue title [CVE: 0000-0000 found in Jinja2 - Version: 2.11.3 [PYTHON]]
Checking for issue title [CVE: 2020-14343 found in PyYAML - Version: 5.3.1 [PYTHON]]

Comment (hovaesco) [reply .github/workflows/security.yml]: `fail-on-cvss: 0` will fail the build or just create issues?

Comment (mdesmet) [reply .github/workflows/security.yml]: Yes it would fail if they found one or more issues. I think we may just omit this parameter (the default is 10).

**Review by hovaesco — APPROVED** [review]

**Pull request merged**